specification/digitaltwins/resource-manager/Microsoft.DigitalTwins/stable/2022-10-31/digitaltwins.json

```diff
@@ -0,0 +1,2299 @@
+          "$ref": "#/definitions/PrivateEndpoint"
+        },
+        "groupIds": {
+          "description": "The list of group ids for the private endpoint connection.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/GroupId"
+          }
+        },
+        "privateLinkServiceConnectionState": {
+          "description": "The connection state.",
+          "allOf": [
+            {
+              "$ref": "#/definitions/ConnectionState"
+            }
+          ]
+        }
+      }
+    },
+    "PrivateEndpointConnection": {
+      "description": "The private endpoint connection of a Digital Twin.",
+      "type": "object",
+      "x-ms-azure-resource": true,
+      "properties": {
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The resource identifier."
+        },
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The resource name.",
+          "pattern": "^(?![0-9]+$)(?!-)[a-zA-Z0-9-]{2,49}[a-zA-Z0-9]$"
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The resource type."
+        },
+        "properties": {
+          "description": "The connection properties.",
+          "$ref": "#/definitions/ConnectionProperties"
+        },
+        "systemData": {
+          "description": "Metadata pertaining to creation and last modification of the private endpoint connection.",
+          "readOnly": true,
+          "x-nullable": true,
+          "$ref": "#/definitions/SystemData"
+        }
+      },
+      "required": [
+        "properties"
+      ]
+    },
+    "TimeSeriesDatabaseConnectionProperties": {
+      "description": "Properties of a time series database connection resource.",
+      "discriminator": "connectionType",
+      "required": [
+        "connectionType"
+      ],
+      "type": "object",
+      "properties": {
+        "connectionType": {
+          "description": "The type of time series connection resource.",
+          "enum": [
+            "AzureDataExplorer"
+          ],
+          "x-ms-enum": {
+            "name": "connectionType",
+            "modelAsString": true
+          },
+          "type": "string"
+        },
+        "provisioningState": {
+          "description": "The provisioning state.",
+          "enum": [
+            "Provisioning",
+            "Deleting",
+            "Updating",
+            "Succeeded",
+            "Failed",
+            "Canceled",
+            "Deleted",
+            "Warning",
+            "Suspending",
+            "Restoring",
+            "Moving",
+            "Disabled"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "TimeSeriesDatabaseConnectionState",
+            "modelAsString": true
+          }
+        },
+        "identity": {
```

Comment:
>identity [](http://example.com/codeflow?start=9&length=8)

Identity is a top level property not meant to be under properties Are you doing some custom implementation ?

Reply:
Yes, this is a custom implementation. This isn't the same as the 'identity' property for DigitalTwinsInstances - TimeSeriesDatabaseConnection is a proxy resource

Reply:
Why do you need the custom implementation ? Proxy resources support MSI as explained here : 
https://armwiki.azurewebsites.net/authorization/managed_identities/MSIOnboardingProxyResource.html

Full details here : 
https://armwiki.azurewebsites.net/authorization/managed_identities/MSIOnboardingSteps.html

Reply:
The intention is not to assign a new identity to a proxy resource, but rather choose an identity associated with a DigitalTwinsInstance, for aunthentication to a customer owned resource. This is similar to what IoTHub has done [here](https://github.com/Azure/azure-rest-api-specs/blob/450b1618259d8b170b5cfcb37866d3fc5747b564/specification/iothub/resource-manager/Microsoft.Devices/stable/2021-07-02/iothub.json#L2518). We already support this functionality for system assigned identities, this change is now extending the identity based authentication to support user assigned identities.

Reply:
Got it , thanks for explaining. Can you please rename the "ManagedIdentity" to something like "ManagedIdentityReference" so that this is clear for future reviewers ? Signing off on this otherwise , please fix before merging.

Reply:
This is resolved!